internal/olm/client/client.go

```diff
@@ -202,3 +208,60 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 
 	return wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
 }
+
+func (c Client) GetInstalledVersion(ctx context.Context) (string, error) {
+	opts := client.InNamespace(OLMNamespace)
+	csvs := &olmapiv1alpha1.ClusterServiceVersionList{}
+	if err := c.KubeClient.List(ctx, opts, csvs); err != nil {
+		if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+			return "", ErrOLMNotInstalled
+		}
+		return "", errors.Wrap(err, "failed to get OLM version from CSVs")
+	}
+	var pkgServerCSV *olmapiv1alpha1.ClusterServiceVersion
+	for _, csv := range csvs.Items {
+		name := csv.GetName()
+		// Check old and new name possibilities.
+		if name == pkgServerCSVNewName || strings.HasPrefix(name, pkgServerCSVOldNamePrefix) {
+			// There is more than one version of OLM installed in the cluster,
+			// so we can't resolve the version being used.
+			if pkgServerCSV != nil {
+				return "", errors.New("failed to get OLM version: more than one OLM version installed")
```

Comment:
Nit(since it's not a part of this refactoring PR): Maybe add the two CSV names and versions to this error message.